PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobLocalRunConfigurationProducer.java

```diff
@@ -48,15 +49,14 @@
 import com.microsoft.azure.hdinsight.spark.run.configuration.RemoteDebugRunConfiguration;
 import com.microsoft.azure.hdinsight.spark.run.configuration.RemoteDebugRunConfigurationType;
 import org.jetbrains.annotations.NotNull;
-import org.jetbrains.plugins.scala.lang.psi.api.statements.ScFunctionDefinition;
+import org.jetbrains.plugins.scala.runner.ScalaMainMethodUtil;
 import scala.Option;
 import scala.Tuple2;
 
-import java.lang.reflect.InvocationTargetException;
-import java.lang.reflect.Method;
 import java.util.AbstractMap.SimpleImmutableEntry;
 import java.util.Optional;
 import java.util.Set;
+import java.util.function.Function;
 
 public class SparkBatchJobLocalRunConfigurationProducer extends JavaRunConfigurationProducerBase<RemoteDebugRunConfiguration> {
```

Comment:
>SparkBatchJobLocalRunConfigurationProducer [](start = 13, length = 42)

Should this be just runconfigurationproducer?

Reply:
Naming this since we only auto generate the run configuration for local run. :) I can fix it.

Reply:
Why not for remote?

---
In reply to: [210156954](https://github.com/Microsoft/azure-tools-for-java/pull/1878#discussion_r210156954) [](ancestors = 210156954)

Reply:
The class is bind to the 'Run' executor, not 'Spark Job Submit' added by us. So, only the local run works.